File: LC_JS-SOLUTIONS/540. Single Element in a Sorted Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNonDuplicate = function (nums) {
  let l = 0;
  let r = nums.length - 1;

  while (l <= r) {
    let m = l + Math.floor((r - l) / 2);

    if (
      (m - 1 < 0 || nums[m] != nums[m - 1]) &&
      (m + 1 === nums.length || nums[m] != nums[m + 1])
    ) {
      return nums[m];
    }

    let leftSide = nums[m] === nums[m - 1] ? m - 1 : m;

    if (leftSide % 2) {
      r = m - 1;
    } else {
      l = m + 1;
    }
  }
};

File: LC-Python-Solution/540. Single Element in a Sorted Array.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        l = 0
        r = len(nums)-1

        while l<=r:
            m = l + ((r-l)//2)

            if((m+1 == len(nums) or nums[m]!=nums[m+1]) and (m-1<0 or nums[m]!=nums[m-1])):
                return nums[m]
            
            leftSize = m-1 if nums[m] ==nums[m-1] else m

            if leftSize%2:
                r = m - 1
            else:
                l = m + 1









        